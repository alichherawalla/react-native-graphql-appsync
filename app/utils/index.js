import { fonts } from 'app/themes'

export const getRandomKey = () =>
  Math.random()
    .toString(36)
    .substr(2, 3)
export const timeout = ms => new Promise(resolve => setTimeout(resolve, ms))

export const rnFontSizeAdapter = type => ({
  fontSize: parseInt(
    fonts.size[type]()[0]
      .replace('font-size: ', '')
      .replace('px;', '')
      .trim(),
    10
  )
})

export const rnFontWeightAdapter = type => ({
  fontSize: parseInt(
    fonts.weights[type]()[0]
      .replace('font-weight: ', '')
      .trim(),
    10
  )
})

export const NEW_EMPLOYEE = {
  employee: {
    address: [{}],
    skills: [{}]
  }
}

export const SIGNUP_CONFIG = {
  header: 'Sign up to buildops',
  hideAllDefaults: true,
  defaultCountryCode: '1',
  signUpFields: [
    {
      label: 'User name',
      key: 'username',
      required: true,
      displayOrder: 1,
      type: 'string'
    },
    {
      label: 'Password',
      key: 'password',
      required: true,
      displayOrder: 2,
      type: 'password'
    },
    {
      label: 'Email',
      key: 'email',
      required: true,
      displayOrder: 4,
      type: 'string'
    }
  ]
}
