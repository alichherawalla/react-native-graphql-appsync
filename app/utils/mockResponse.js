export const employeeData = [
  {
    id: 1,
    firstname: 'John',
    lastname: 'Doe',
    address: [
      {
        line1: '3014 Dauphine',
        line2: 'St Ste A',
        city: 'New Orleans',
        state: 'LA',
        zipcode: '70117'
      },
      {
        line1: '3020 Dauphine',
        line2: 'St Ste A',
        city: 'New Orleans',
        state: 'LA',
        zipcode: '70117'
      }
    ],
    skills: [
      { id: 1, name: 'Technician' },
      { id: 2, name: 'Plumber' },
      { id: 3, name: 'Electrician' }
    ]
  },
  {
    id: 2,
    firstname: 'Jonathan',
    lastname: 'Doe',
    address: [
      {
        line1: '3022 Dauphine',
        line2: 'St Ste A',
        city: 'New Orleans',
        state: 'LA',
        zipcode: '70117'
      },
      {
        line1: '3024 Dauphine',
        line2: 'St Ste A',
        city: 'New Orleans',
        state: 'LA',
        zipcode: '70117'
      },
      {
        line1: '3026 Dauphine',
        line2: 'St Ste A',
        city: 'New Orleans',
        state: 'LA',
        zipcode: '70117'
      },
      {
        line1: '3028 Dauphine',
        line2: 'St Ste A',
        city: 'New Orleans',
        state: 'LA',
        zipcode: '70117'
      }
    ],
    skills: [{ id: 6, name: 'Senior Electrician' }]
  },
  {
    id: 3,
    firstname: 'Steve',
    lastname: 'Doe',
    address: [
      {
        line1: '3028 Dauphine',
        line2: 'St Ste A',
        city: 'New Orleans',
        state: 'LA',
        zipcode: '70117'
      }
    ],
    skills: [
      { id: 4, name: 'Senior Technician' },
      { id: 5, name: 'Senior Plumber' }
    ]
  },
  {
    id: 4,
    firstname: 'Nathan',
    lastname: 'Doe',
    address: [
      {
        line1: '3022 Dauphine',
        line2: 'St Ste A',
        city: 'New Orleans',
        state: 'LA',
        zipcode: '70117'
      },
      {
        line1: '3024 Dauphine',
        line2: 'St Ste A',
        city: 'New Orleans',
        state: 'LA',
        zipcode: '70117'
      },
      {
        line1: '3026 Dauphine',
        line2: 'St Ste A',
        city: 'New Orleans',
        state: 'LA',
        zipcode: '70117'
      }
    ],
    skills: [
      { id: 4, name: 'Senior Technician' },
      { id: 5, name: 'Senior Plumber' },
      { id: 6, name: 'Senior Electrician' }
    ]
  },
  {
    id: 5,
    firstname: 'Albert',
    lastname: 'Doe',
    address: [
      {
        line1: '3022 Dauphine',
        line2: 'St Ste A',
        city: 'New Orleans',
        state: 'LA',
        zipcode: '70117'
      },
      {
        line1: '3024 Dauphine',
        line2: 'St Ste A',
        city: 'New Orleans',
        state: 'LA',
        zipcode: '70117'
      },
      {
        line1: '3026 Dauphine',
        line2: 'St Ste A',
        city: 'New Orleans',
        state: 'LA',
        zipcode: '70117'
      },
      {
        line1: '3028 Dauphine',
        line2: 'St Ste A',
        city: 'New Orleans',
        state: 'LA',
        zipcode: '70117'
      }
    ],
    skills: [
      { id: 4, name: 'Senior Technician' },
      { id: 5, name: 'Senior Plumber' },
      { id: 6, name: 'Senior Electrician' }
    ]
  }
]
