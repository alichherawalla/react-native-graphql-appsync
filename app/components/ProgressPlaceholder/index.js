/**
 *
 * ProgressPlaceholder
 *
 */

import React from 'react'
import { ProgressBar, Text } from 'react-native-paper'
import { StyleSheet, View } from 'react-native'
import PropTypes from 'prop-types'
import Container from 'app/components/Container'
import If from 'app/components/If'
import { rnFontSizeAdapter } from 'app/utils'
import { colors } from 'app/themes'

function ProgressPlaceholder({ placeholderText, show }) {
  return (
    <If condition={show}>
      <Container testID="progress-placeholder" style={styles.container}>
        <ProgressBar indeterminate color="blue" />
        <View>
          <Text style={styles.textStyle}>{placeholderText}</Text>
        </View>
      </Container>
    </If>
  )
}

const styles = StyleSheet.create({
  container: {
    justifyContent: 'space-between',
    flex: 1,
    display: 'flex'
  },
  textStyle: {
    ...rnFontSizeAdapter('big'),
    textAlign: 'center',
    color: colors.gray
  }
})
ProgressPlaceholder.propTypes = {
  placeholderText: PropTypes.string.isRequired,
  show: PropTypes.bool
}

export default ProgressPlaceholder
