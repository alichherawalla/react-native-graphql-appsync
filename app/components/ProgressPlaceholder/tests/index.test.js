/**
 *
 * Tests for ProgressPlaceholder
 *
 */

import React from 'react'
// import { fireEvent } from '@testing-library/dom'
import { renderWithIntl } from 'app/utils/testUtils'
import ProgressPlaceholder from '../index'

describe('<ProgressPlaceholder />', () => {
  it('should render and match the snapshot', () => {
    const { baseElement } = renderWithIntl(<ProgressPlaceholder show />)
    expect(baseElement).toMatchSnapshot()
  })

  it('should contain 1 ProgressPlaceholder component', () => {
    const { getAllByTestId } = renderWithIntl(<ProgressPlaceholder show />)
    expect(getAllByTestId('progress-placeholder').length).toBe(1)
  })
})
