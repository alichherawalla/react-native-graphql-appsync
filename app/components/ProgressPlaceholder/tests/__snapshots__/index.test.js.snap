// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProgressPlaceholder /> should render and match the snapshot 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <SafeAreaView
      style={
        Array [
          Object {
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
          },
          Object {
            "display": "flex",
            "flex": 1,
            "justifyContent": "space-between",
          },
        ]
      }
      testID="progress-placeholder"
    >
      <View>
        <View
          style={
            Object {
              "backgroundColor": "rgba(0, 0, 255, 0.38)",
              "height": 4,
              "opacity": 0,
              "overflow": "hidden",
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "blue",
                "flex": 1,
                "transform": Array [
                  Object {
                    "translateX": -0,
                  },
                  Object {
                    "scaleX": 0.0001,
                  },
                ],
                "width": 0,
              }
            }
          />
        </View>
      </View>
      <View>
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
                "textAlign": "left",
              },
              Object {
                "color": "#d3d3d3",
                "fontSize": 20,
                "textAlign": "center",
              },
            ]
          }
        />
      </View>
    </SafeAreaView>
  </View>
</View>
`;
